// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package privacy_sandbox.server_common.gvisor;


// Integrator teams will declare this interface message
// Roma untrusted binary is expected to deserialize this and get the BinaryRequest
message BinaryRequest {
  Function function = 1;
  enum Function {
    FUNCTION_UNSPECIFIED = 0;
    FUNCTION_HELLO_WORLD = 1;
    FUNCTION_PRIME_SIEVE = 2;
    FUNCTION_CALLBACK = 3;
    FUNCTION_TEN_CALLBACK_INVOCATIONS = 4;
    FUNCTION_CONCURRENT_CALLBACKS = 5;
  }
}

// Integrator teams will declare this interface message
// Roma untrusted binary is expected to provide serialized version of this as BinaryResponse
message BinaryResponse {
  bytes greeting = 1;
  repeated int32 prime_number = 2;
}
